docs/_newsfragments/2063.feature.rst

```diff
@@ -0,0 +1 @@
+A new property, :attr:`~falcon.Request.headers_lower`, was added to provide a unified, self-documenting way to get a copy of all request headers with lower-cased names to facilitate case-insensitive matching for middleware components, etc. that need to be compatible with both WSGI and ASGI. This was done in lieu of introducing a breaking change to the WSGI :attr:`~falcon.Request.headers` property which currently returns uppercased headers from the WSGI environ dictionary.
```

Comment:
Maybe we should put a comma after that `etc.` :arrow_right: `etc.,`, because now it feels like the end of a sentence otherwise?

Reply:
(This one may be just me failing to read it the right way.)

Reply:
I reworked the prose to reduce sentence length, and removed the `etc.` since it wasn't really adding anything.